utils.py

```diff
@@ -0,0 +1,20 @@
+from functools import wraps
+from flask import jsonify, request
+import documents.state_machine as state_machine
+
+def authenticate_user(func):
+    @wraps(func)
+    def wrapper():
+        auth_header = request.headers.get('Authorization')
+        payload = state_machine.token_valid_check(auth_header)
+        return func(payload)
```

Comment:
same comment as before: will not work for APIs with path params

Reply:
Didn't get it before too, can't find the bug here

Reply:
It's not working on the GET request too but can't understand what's going wrong here